lib/bridge/pbn/game_parser.rb

```diff
@@ -0,0 +1,53 @@
+module Bridge
+  module Pbn
+    class GameParser
+      include SingleCharComparisonConstants
+
+      # TODO: enforce 255 char cap on line width
+      def each_subgame(pbn_game_string, &block)
+        @pbn_game_string = pbn_game_string
+        @builder = SubgameBuilder.new
+        @state = GameParserStates::BeforeFirstTag.new(self, @builder)
+        @block = block
+        @builder.clear
+        process
+      end
+
+      def process
+        @pbn_game_string.each_char.with_index do |char, index|
+          @cur_char_index = index
```

Comment:
Thanks for the feedback. I'm trying to hide cur_char_index from the states, since it isn't in their domain of concern. The usage of char_index is in reporting when a state sends raise_error back to the parser, which is (currently) this class's domain of concern. Is there a better design here? And if not, is there a clearer way of expressing it?